ddtrace/llmobs/_llmobs.py

```diff
@@ -774,6 +777,153 @@ def _tag_metrics(span: Span, metrics: Dict[str, Any]) -> None:
+        if cls.enabled is False:
+            log.warning(
+                "LLMObs.submit_evaluation_for() called when LLMObs is not enabled. ",
+                "Evaluation metric data will not be sent.",
+            )
+            return
+        if not config._dd_api_key:
+            log.warning(
+                "DD_API_KEY is required for sending evaluation metrics. Evaluation metric data will not be sent. "
+                "Ensure this configuration is set before running your application."
+            )
+            return
+
+        has_exactly_one_joining_key = (span is not None) ^ (span_with_tag is not None)
+
+        if not has_exactly_one_joining_key:
+            log.warning("Exactly one of `span` or `span_with_tag` must be specified to submit an evaluation metric.")
+            return
+
+        join_on = {}
+        if span is not None:
+            if (
+                not isinstance(span, dict)
+                or "span_id" not in span
+                or "trace_id" not in span
+                or not isinstance(span.get("span_id"), str)
+                or not isinstance(span.get("trace_id"), str)
+            ):
+                log.warning(
+                    "`span` must be a dictionary containing both span_id and trace_id keys. "
+                    "LLMObs.export_span() can be used to generate this dictionary from a given span."
+                )
+                return
+            join_on["span"] = span
+        elif span_with_tag is not None:
+            if (
+                not isinstance(span_with_tag, tuple)
+                or len(span_with_tag) != 2
+                or not all(isinstance(i, str) for i in span_with_tag)
+            ):
+                log.warning("`span_with_tag` must be a tuple of shape (tag_key, tag_value)")
+                return
+            join_on["tag"] = {"tag_key": span_with_tag[0], "tag_value": span_with_tag[1]}
+
+        ml_app = ml_app if ml_app else config._llmobs_ml_app
+        if not ml_app:
+            log.warning(
+                "ML App name is required for sending evaluation metrics. Evaluation metric data will not be sent. "
+                "Ensure this configuration is set before running your application."
+            )
+            return
+
+        timestamp_ms = timestamp_ms if timestamp_ms else int(time.time() * 1000)
+
+        if not isinstance(timestamp_ms, int) or timestamp_ms < 0:
+            log.warning("timestamp_ms must be a non-negative integer. Evaluation metric data will not be sent")
+            return
+
+        if not label:
+            log.warning("label must be the specified name of the evaluation metric.")
+            return
+
+        if not metric_type or metric_type.lower() not in ("categorical", "numerical", "score"):
+            log.warning("metric_type must be one of 'categorical' or 'score'.")
+            return
+
+        metric_type = metric_type.lower()
+        if metric_type == "numerical":
+            log.warning(
```

Comment:
do we have to do this backwards support since this is a new method? This is just cause we were doing it in the other eval method right?

Reply:
We don't need to support this; I just missed that i needed to remove it while copying the logic from the old method, thx